### ISSUE TITLE
in plot, legend = F does not suppress legend

### ISSUE THREAD

**Issue by ptitle** [rspatial/terra#746]
Based on the documentation, my expectation is that the following would plot the raster without a legend:

```
library(terra)
f <- system.file("ex/elev.tif", package="terra") 
r <- rast(f)
plot(r, legend = FALSE)
```
But the legend gets plotted regardless.

**Comment by ptitle** [comment]
Apologies, this appears to be fixed on Github.

**Comment by kadyb** [comment]
Just for reference, this was reported and fixed earlier in #738.
